src/transaction/id.rs

```diff
@@ -1,74 +1,76 @@
-use crate::{Input, Metadata, Output, Transaction, UtxoId};
+use crate::{Input, Metadata, Output, Transaction};
 
 use fuel_crypto::Hasher;
 use fuel_types::bytes::SerializableVec;
 use fuel_types::Bytes32;
 
+use core::mem;
+
 impl Transaction {
     pub(crate) fn prepare_sign(&mut self) -> &mut Self {
         self.set_receipts_root(Default::default());
 
-        let (inputs, outputs, witnesses) = match self {
-            Self::Create {
-                inputs,
-                outputs,
-                witnesses,
-                ..
-            } => (inputs, outputs, witnesses),
-            Self::Script {
-                inputs,
-                outputs,
-                witnesses,
-                ..
-            } => (inputs, outputs, witnesses),
-        };
+        self._inputs_mut().iter_mut().for_each(|input| match input {
+            Input::CoinSigned { tx_pointer, .. } | Input::CoinPredicate { tx_pointer, .. } => {
+                mem::take(tx_pointer);
+            }
 
-        inputs.iter_mut().for_each(|input| {
-            if let Input::Contract {
+            Input::Contract {
                 utxo_id,
                 balance_root,
                 state_root,
+                tx_pointer,
                 ..
-            } = input
-            {
-                *utxo_id = UtxoId::default();
-                balance_root.iter_mut().for_each(|b| *b = 0);
-                state_root.iter_mut().for_each(|b| *b = 0);
+            } => {
+                mem::take(utxo_id);
```

Comment:
nit: when not using the return value of `mem::take`, I find using `*value = Default::default();` to be more readable

Reply:
isn't the shorter the better? I like `mem::take(utxo_id);` because of the reduced verbosity, and the std api is clear on what it is doing (e.g. taking the value)

Reply:
Explict is better, `take` is intended for a specific situation. Docs are clear, yes, but with =default I wouldn't have to check the docs at all. Anyway, if you feel like this is better, we can just go with it.